### ISSUE TITLE
Use the same colors in Scrollbar and Textarea

### ISSUE THREAD

**Issue by callamaris** [ryu1kn/vscode-text-marker#12]
Hi,

can you please add an option to use the same colors in the scrolbar and in the textarea?

**Comment by ryu1kn** [comment]
Hi @callamaris,

I also thought about it a while back but didn't do it as it could be confusing if you have multiple colours on the same line in the text area.

Let's say you have yellow on line 10 and 100 and blue on line 5 and 100. On the scrollbar, TextMarker may show only 1 yellow for line 10 and 2 blues for line 5 and 100. If you look at the scrollbar, you would think yellow is only on line 10, and don't notice the other yellow on line 100...

If you have an idea to get around with this issue, let me know!!

**Comment by ryu1kn** [comment]
Now I think this again, maybe we introduce this and let the user to decide whether they want to use the same colours on the scrollbar providing a new config like `useSameColorOnScrollbar`

**Comment by callamaris** [comment]
in case of multiple colors in the same line it would be great, when its configurable, which color is shown in the scrollbar.

possible options: longest, shortest, oldest, latest or first/last highlighted character in line

the most intuitive default option, i think, would be when the latest created highlighting wins.

**Comment by ryu1kn** [comment]
Released as v1.0.0 :+1: I introduced a new config which enables you to use the highlight colours on the ruler. In case of multiple colours on the same line, the ruler colour depends on how the editor works; hence it's a default off option.

```
* `textmarker.useHighlightColorOnRuler` (default: `false`)

    Use the same color for both highlighting a text and showing the location on the ruler.
    If you have multiple highlights on the same line, the corresponding ruler color will be one of theirs.
```

https://github.com/ryu1kn/vscode-text-marker/blob/master/README.md#extension-settings
